Build failure: uucp-1.07

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

run 'nix-shell -p uucp' 

see it failing when compiling uuconf subdir. Since uucp hasn't changed in the past 8 month I guess some dependency is failing. 


### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/319181359

### Relevant log output

```console
gcc -DHAVE_CONFIG_H -I. -I..    -I.. -I./.. -W -Wall -Wstrict-prototypes -Wmissing-prototypes -DNEWCONFIGLIB=\"/nix/store/wcjf9rl61sqnl7nsg5h3ryk9k1s571p7-uucp-1.07/conf/uucp\" -DOLDCONFIGLIB=\"/usr/lib/uucp\" -g -O2 -c -o cmdarg.o cmdarg.c
cmdarg.c:37:12: error: conflicting types for 'strcmp'; have 'int(void)'
   37 | extern int strcmp (), strcasecmp ();
      |            ^~~~~~
In file included from /nix/store/dj43clc5ff7jjnfmhbaj6q4q0h8kpfpm-glibc-2.40-66-dev/include/memory.h:29,
                 from ../uucp.h:197,
                 from uucnfi.h:29,
                 from cmdarg.c:25:
/nix/store/dj43clc5ff7jjnfmhbaj6q4q0h8kpfpm-glibc-2.40-66-dev/include/string.h:156:12: note: previous declaration of 'strcmp' with type 'int(const char *, const char *)'
  156 | extern int strcmp (const char *__s1, const char *__s2)
      |            ^~~~~~
cmdarg.c:37:23: error: conflicting types for 'strcasecmp'; have 'int(void)'
   37 | extern int strcmp (), strcasecmp ();
      |                       ^~~~~~~~~~
In file included from /nix/store/dj43clc5ff7jjnfmhbaj6q4q0h8kpfpm-glibc-2.40-66-dev/include/string.h:462:
/nix/store/dj43clc5ff7jjnfmhbaj6q4q0h8kpfpm-glibc-2.40-66-dev/include/strings.h:116:12: note: previous declaration of 'strcasecmp' with type 'int(const char *, const char *)'
  116 | extern int strcasecmp (const char *__s1, const char *__s2)
      |            ^~~~~~~~~~
cmdarg.c: In function 'uuconf_cmd_args':
cmdarg.c:42:1: warning: old-style function definition [-Wold-style-definition]
   42 | uuconf_cmd_args (pglobal, cargs, pzargs, qtab, pinfo, pfiunknown, iflags,
      | ^~~~~~~~~~~~~~~
cmdarg.c:62:12: error: assignment to 'int (*)(const char *, const char *)' from incompatible pointer type 'int (*)(void)' [-Wincompatible-pointer-types]
   62 |     pficmp = strcmp;
      |            ^
cmdarg.c:37:12: note: 'strcmp' declared here
   37 | extern int strcmp (), strcasecmp ();
      |            ^~~~~~
cmdarg.c:69:14: error: assignment to 'int (*)(const char *, const char *)' from incompatible pointer type 'int (*)(void)' [-Wincompatible-pointer-types]
   69 |       pficmp = strcasecmp;
      |              ^
cmdarg.c:37:23: note: 'strcasecmp' declared here
   37 | extern int strcmp (), strcasecmp ();
      |                       ^~~~~~~~~~
make[2]: *** [Makefile:503: cmdarg.o] Error 1
make[2]: Leaving directory '/build/uucp-1.07/uuconf'
make[1]: *** [Makefile:1150: all-recursive] Error 1
make[1]: Leaving directory '/build/uucp-1.07'
make: *** [Makefile:540: all] Error 2
error: Cannot build '/nix/store/pqxm1h4fdrdy2s9xi6sbww5zvc5v4hzi-uucp-1.07.drv'.
       Reason: builder failed with exit code 2.
       Output paths:
         /nix/store/wcjf9rl61sqnl7nsg5h3ryk9k1s571p7-uucp-1.07
       Last 25 log lines:
       > In file included from /nix/store/dj43clc5ff7jjnfmhbaj6q4q0h8kpfpm-glibc-2.40-66-dev/include/string.h:462:
       > /nix/store/dj43clc5ff7jjnfmhbaj6q4q0h8kpfpm-glibc-2.40-66-dev/include/strings.h:116:12: note: previous declaration of 'strcasecmp' with type 'int(const char *, const char *)'
       >   116 | extern int strcasecmp (const char *__s1, const char *__s2)
       >       |            ^~~~~~~~~~
       > cmdarg.c: In function 'uuconf_cmd_args':
       > cmdarg.c:42:1: warning: old-style function definition [-Wold-style-definition]
       >    42 | uuconf_cmd_args (pglobal, cargs, pzargs, qtab, pinfo, pfiunknown, iflags,
       >       | ^~~~~~~~~~~~~~~
       > cmdarg.c:62:12: error: assignment to 'int (*)(const char *, const char *)' from incompatible pointer type 'int (*)(void)' [-Wincompatible-pointer-types]
       >    62 |     pficmp = strcmp;
       >       |            ^
       > cmdarg.c:37:12: note: 'strcmp' declared here
       >    37 | extern int strcmp (), strcasecmp ();
       >       |            ^~~~~~
       > cmdarg.c:69:14: error: assignment to 'int (*)(const char *, const char *)' from incompatible pointer type 'int (*)(void)' [-Wincompatible-pointer-types]
       >    69 |       pficmp = strcasecmp;
       >       |              ^
       > cmdarg.c:37:23: note: 'strcasecmp' declared here
       >    37 | extern int strcmp (), strcasecmp ();
       >       |                       ^~~~~~~~~~
       > make[2]: *** [Makefile:503: cmdarg.o] Error 1
       > make[2]: Leaving directory '/build/uucp-1.07/uuconf'
       > make[1]: *** [Makefile:1150: all-recursive] Error 1
       > make[1]: Leaving directory '/build/uucp-1.07'
       > make: *** [Makefile:540: all] Error 2
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.64, NixOS, 26.05 (Yarara), 26.05pre925861.ffbc9f8cbaac`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2+2`
 - channels(root): `"nixos, nixos-unstable"`
 - nixpkgs: `/nix/store/9yzh0hf9mq844amfsypl08d8ywysksv1-nixos/nixos`


### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/uu/uucp/package.nix
+++ b/pkgs/by-name/uu/uucp/package.nix
@@ -2,6 +2,7 @@
   lib,
   stdenv,
   fetchurl,
+  fetchDebianPatch,
   autoreconfHook,
   testers,
 }:
@@ -22,14 +23,28 @@ stdenv.mkDerivation (finalAttrs: {
     substituteInPlace Makefile.am \
       --replace-fail 4555 0555
     sed -i '/chown $(OWNER)/d' Makefile.am
-
-    # don't reply on implicitly defined `exit` function in `HAVE_VOID` test:
-    substituteInPlace configure.in \
-      --replace-fail '(void) exit (0)' '(void) (0)'
   '';
 
   patches = [
     ./socklen_t.patch
+    (fetchDebianPatch {
+      inherit (finalAttrs) pname version;
+      debianRevision = "31";
+      patch = "configure.patch";
+      hash = "sha256-6Aqghz6P+bWULHOXCQIdQLRuaE+Lci7t5ojQXJOyeA0=";
+    })
+    (fetchDebianPatch {
+      inherit (finalAttrs) pname version;
+      debianRevision = "31";
+      patch = "implicit.patch";
+      hash = "sha256-EsJqZCV4x7ggzpoa4OaibCLvF8L8FGGnLlBtr4Cee18=";
+    })
+    (fetchDebianPatch {
+      inherit (finalAttrs) pname version;
+      debianRevision = "31";
+      patch = "gcc15.patch";
+      hash = "sha256-+9H/gQLwkPx4GeWiZyy6oQdysvw2+P1O8wP5It/Tg5k=";
+    })
   ];
 
   # Regenerate `configure`; the checked in version was generated in 2002 and

